x/genutil/src/gentx.rs

```diff
@@ -0,0 +1,234 @@
+use std::{net::SocketAddr, path::PathBuf};
+
+use gears::{
+    application::handlers::client::TxHandler,
+    commands::client::tx::{ClientTxContext, TxCommand},
+    crypto::{ed25519::Ed25519PubKey, public::PublicKey},
+    types::{
+        base::{coin::UnsignedCoin, coins::UnsignedCoins},
+        decimal256::Decimal256,
+        tx::Messages,
+        uint::Uint256,
+    },
+};
+use staking::{CommissionRates, CreateValidator, Description};
+
+use crate::utils::{parse_staking_params_from_genesis, GenesisAccounts, GenesisBalanceIter};
+
+use gears::tendermint::types::proto::crypto::PublicKey as TendermintPublicKey;
+
+#[derive(Debug, Clone)]
+pub struct GentxCmd {
+    pub pubkey: Option<TendermintPublicKey>,
+    pub amount: UnsignedCoin,
+    pub moniker: String,
+    pub identity: String,
+    pub website: String,
+    pub security_contact: String,
+    pub details: String,
+    pub commission_rate: Decimal256,
+    pub commission_max_rate: Decimal256,
+    pub commission_max_change_rate: Decimal256,
+    pub min_self_delegation: Uint256,
+
+    pub output: Option<PathBuf>,
+    pub ip: SocketAddr,
+    pub node_id: Option<String>,
+}
+
+pub fn gentx_cmd(
+    cmd: TxCommand<GentxCmd>,
+    balance_sk: &'static str,
+    staking_sk: &'static str,
+    auth_sk: &'static str,
+) -> anyhow::Result<()> {
+    let gentx_handler =
+        GentxTxHandler::new(cmd.inner.output.clone(), balance_sk, staking_sk, auth_sk)?;
+
+    gears::commands::client::tx::run_tx(cmd, &gentx_handler)?;
+
+    Ok(())
+}
+
+#[derive(Debug, Clone)]
+struct GentxTxHandler {
+    output_dir: Option<PathBuf>,
+    pub balance_sk: &'static str,
+    pub staking_sk: &'static str,
+    pub auth_sk: &'static str,
+}
+
+impl GentxTxHandler {
+    pub fn new(
+        output_dir: Option<PathBuf>,
```

Comment:
It should default to using a `gentx` directory in the `config` directory.

Reply:
Edited cli